GVFS/GVFS.Virtualization/BlobSize/BlobSizes.cs

```diff
@@ -333,6 +334,11 @@ public BlobSizesConnection(BlobSizes blobSizes, string connectionString)
                     this.querySizeCommand.CommandText = "SELECT size FROM BlobSizes WHERE sha = (@sha);";
                     this.querySizeCommand.Parameters.Add(this.shaParam);
                     this.querySizeCommand.Prepare();
+
+                    this.getAllDataCommand = this.connection.CreateCommand();
+
+                    this.getAllDataCommand.CommandText = "SELECT sha, size FROM BlobSizes;";

```

Comment:
This has the possibility of returning several million rows in what I believe would not be a very uncommon case in some of Microsoft's internal repositories. We need to limit this query to only what we need to project the index, but pulling that data out in a batch, or several, might be a feasible optimization. Have you explored that?

Reply:
Not yet - went with the simple option first and in my case (~100k placeholders) it worked well. Let me see how this would work, if we specified the sha's of the items, that we're interested in.

Reply:
Thanks! I think this could certainly help performance, if we can tune the batch size correctly.